src/java/org/apache/cassandra/concurrent/AbstractLocalAwareExecutorService.java

```diff
@@ -150,7 +149,7 @@ class FutureTask<T> extends SimpleCondition implements Future<T>, Runnable
 
         public FutureTask(Callable<T> callable)
```

Comment:
Seems like those get created for a wide variety of tasks in Cassandra. Did we get a chance to verify from a rc the log volume + perf from adding this additional wrapper? 
The Tracers metadata will be used for all traceIds coming from Alta correct? Or am I mixing up?

Reply:
So for any tasks generated that's not part of a thrift request, a random traceId will be used, which will be subjected to the configured trace rate (intentionally very low for us). Additionally, we have the ability to downsample forced client requests, so we can continue to reduce our trace volume if we need to. 

This on it's own won't add too many spans (just the one for the task executing), and in the cases where we're not sampling a trace (~99% of the time), the performance hit should be negligible (an `Unsampled` span has very cheap implementations of all trace operations).